lib/api/model/model.dart

```diff
@@ -698,3 +710,79 @@ class DmMessage extends Message {
   @override
   Map<String, dynamic> toJson() => _$DmMessageToJson(this);
 }
+
+enum MessageEditState {
+  none,
+  edited,
+  moved;
+
+  // Adapted from the shared code:
+  //   https://github.com/zulip/zulip/blob/1fac99733/web/shared/src/resolved_topic.ts
+  // The canonical resolved-topic prefix.
+  static const String _resolvedTopicPrefix = '✔ ';
+
+  /// Whether the two topics differs and only differs by whether one of them is
+  /// resolved or not.
+  ///
+  /// When a topic is resolved, the clients agree on adding a ✔ prefix to the
+  /// topic string. Topics whose only difference is the ✔ prefix are considered
+  /// the same. This helper can be helpful when checking if a message has been
+  /// moved.
+  static bool topicsDifferByResolvednessOnly(String topic, String prevTopic) {
```

Comment:
This name and documentation describe how the function is implemented; but the way I'd like to think of what it *means* is a bit higher-level. From https://github.com/zulip/zulip-flutter/pull/750#discussion_r1661612507 :
> I think what this method actually is about is: does this topic move reflect a resolve or unresolve operation.

So here's a version based on that:
```suggestion
  /// Whether the given topic move reflected either a "resolve topic"
  /// or "unresolve topic" operation.
  ///
  /// The Zulip "resolved topics" feature is implemented by renaming the topic;
  /// but for purposes of [Message.editState], we want to ignore such renames.
  /// This method identifies topic moves that should be ignored in that context.
  static bool topicMoveWasResolveOrUnresolve(String topic, String prevTopic) {
```

The name is kind of wordy but I couldn't think of a shorter one that was accurate. And `topicsDifferByResolvednessOnly` happens to be the exact same number of characters :slightly_smiling_face: